lib/timezone/timezone.ex

```diff
@@ -406,63 +406,26 @@ defmodule Timex.Timezone do
     difference = diff(date, tz)
     # Offset the provided date's time by the difference
     {seconds_from_zeroyear, microsecs} = do_shift(date, difference)
-    case resolve(name, seconds_from_zeroyear) do
-      {:error, _} = err -> err
-      ^tz ->
-        {{y,m,d},{h,mm,s}} = :calendar.gregorian_seconds_to_datetime(seconds_from_zeroyear)
-        %DateTime{:year => y, :month => m, :day => d,
-                  :hour => h, :minute => mm, :second => s,
-                  :microsecond => microsecs,
-                  :time_zone => tz.full_name,
-                  :zone_abbr => tz.abbreviation,
-                  :utc_offset => tz.offset_utc,
-                  :std_offset => tz.offset_std}
-      %TimezoneInfo{} = new_zone ->
-        difference = diff(tz, new_zone)
-        {shifted, microsecs} = do_shift(seconds_from_zeroyear, microsecs, difference)
-        {{y,m,d},{h,mm,s}} = :calendar.gregorian_seconds_to_datetime(shifted)
-        %DateTime{:year => y, :month => m, :day => d,
-                  :hour => h, :minute => mm, :second => s,
-                  :microsecond => microsecs,
-                  :time_zone => new_zone.full_name,
-                  :zone_abbr => new_zone.abbreviation,
-                  :utc_offset => new_zone.offset_utc,
-                  :std_offset => new_zone.offset_std}
-      %AmbiguousTimezoneInfo{:before => before_tz, :after => after_tz} ->
-        before_diff = diff(tz, before_tz)
-        {before_shifted, before_us} = do_shift(seconds_from_zeroyear, microsecs, before_diff)
-        after_diff = diff(tz, after_tz)
-        {after_shifted, after_us} = do_shift(seconds_from_zeroyear, microsecs, after_diff)
-        {{y,m,d},{h,mm,s}} = :calendar.gregorian_seconds_to_datetime(before_shifted)
-        before_dt = %DateTime{:year => y, :month => m, :day => d,
-                              :hour => h, :minute => mm, :second => s,
-                              :microsecond => before_us,
-                              :time_zone => before_tz.full_name,
-                              :zone_abbr => before_tz.abbreviation,
-                              :utc_offset => before_tz.offset_utc,
-                              :std_offset => before_tz.offset_std}
-        {{y,m,d},{h,mm,s}} = :calendar.gregorian_seconds_to_datetime(after_shifted)
-        after_dt = %DateTime{:year => y, :month => m, :day => d,
-                             :hour => h, :minute => mm, :second => s,
-                             :microsecond => after_us,
-                             :time_zone => after_tz.full_name,
-                             :zone_abbr => after_tz.abbreviation,
-                             :utc_offset => after_tz.offset_utc,
-                             :std_offset => after_tz.offset_std}
-        %AmbiguousDateTime{:before => before_dt, :after => after_dt}
-    end
+    {{y,m,d},{h,mm,s}} = :calendar.gregorian_seconds_to_datetime(seconds_from_zeroyear)
```

Comment:
This doesn't account for cases where the shift is unambiguously in a different variant of the provided timezone, i.e. you want CST (America/Chicago) but when you shift the date/time by the offset, the time it actually represents is CDT (daylight savings time for the same zone).

Additionally, we should verify that when we get an AmbiguousTimezoneInfo, that the provided TimezoneInfo is present in one of the branches; if it is not, we should return the AmbiguousDateTime because we cannot make a choice. If it is present, then we can assume the intent of the developer is that they get a DateTime in the provided TimezoneInfo's zone.